src/commonMain/kotlin/fr/acinq/lightning/channel/InteractiveTx.kt

```diff
@@ -503,31 +641,52 @@ data class InteractiveTxSession(
         previousTxs
     )
 
-    val isComplete: Boolean = txCompleteSent && txCompleteReceived
+    val isComplete: Boolean = txCompleteSent != null && txCompleteReceived != null
 
     fun send(): Pair<InteractiveTxSession, InteractiveTxSessionAction> {
         return when (val msg = toSend.firstOrNull()) {
             null -> {
-                val txComplete = TxComplete(fundingParams.channelId)
-                val next = copy(txCompleteSent = true)
+                // generate a new secret nonce for each musig2 new swapin every time we send TxComplete
```

Comment:
Why do we do that? It is much simpler to generate one secret nonce whenever we send `tx_add_input`, and here to just send the public part, isn't it?

It would also be more consistent since that's what we do for remote inputs: we generate our secret nonce as soon as we receive their swap-in input.

Then you can trivially filter the local secret nonces from the remote secret nonces by just looking at the parity of the `serial_id`.

Reply:
done in https://github.com/ACINQ/lightning-kmp/pull/563/commits/4237120872025646386ada61fe5d2f7f9d359538